In [2]:
from scipy.optimize import minimize

fun = lambda x: (x[0]-x[1])**2 + (x[1]+x[2]-2)**2 + (x[3]-1)**2 + (x[4]-1)**2
const = ({'type': 'eq', 'fun': lambda x: x[0]+3*x[1]},
         {'type': 'eq', 'fun': lambda x: x[2]+x[3]-2*x[4]},
         {'type': 'eq', 'fun': lambda x: x[1]-x[4]})
bounds = ((-10,10),(-10,10),(-10,10),(-10,10),(-10,10))
res = minimize(fun, (-1,2,-3,-4,5), method = 'SLSQP', bounds = bounds, constraints = const)

print(res)

     fun: 4.093023256421694
     jac: array([-2.04652327, -0.18606395, -2.23258728, -2.23252022, -1.48836917])
 message: 'Optimization terminated successfully'
    nfev: 48
     nit: 8
    njev: 8
  status: 0
 success: True
       x: array([-0.76744624,  0.25581541,  0.62789095, -0.11626012,  0.25581541])


In [2]:
from scipy.optimize import minimize

fun = lambda x: (x[0]-x[1])**2 + (x[1]+x[2]-2)**2 + (x[3]-1)**2 + (x[4]-1)**2
const = ({'type': 'eq', 'fun': lambda x: x[0]+3*x[1]},
         {'type': 'eq', 'fun': lambda x: x[2]+x[3]-2*x[4]},
         {'type': 'eq', 'fun': lambda x: x[1]-x[4]})
bounds = ((-10,10),(-10,10),(-10,10),(-10,10),(-10,10))
res = minimize(fun, (-2,6,9,-7,1), method = 'SLSQP', bounds = bounds, constraints = const)

print(res)

     fun: 4.0930232616301065
     jac: array([-2.04648948, -0.18597054, -2.23246008, -2.23267275, -1.48837763])
 message: 'Optimization terminated successfully'
    nfev: 37
     nit: 6
    njev: 6
  status: 0
 success: True
       x: array([-0.76743357,  0.25581119,  0.62795875, -0.11633638,  0.25581119])


In [3]:
from scipy.optimize import minimize

fun = lambda x: (x[0]-x[1])**2 + (x[1]+x[2]-2)**2 + (x[3]-1)**2 + (x[4]-1)**2
const = ({'type': 'eq', 'fun': lambda x: x[0]+3*x[1]},
         {'type': 'eq', 'fun': lambda x: x[2]+x[3]-2*x[4]},
         {'type': 'eq', 'fun': lambda x: x[1]-x[4]})
bounds = ((-10,10),(-10,10),(-10,10),(-10,10),(-10,10))
res = minimize(fun, (9,-6,-9,4,2), method = 'SLSQP', bounds = bounds, constraints = const)

print(res)

     fun: 4.093023255813954
     jac: array([-2.04651159, -0.18604648, -2.23255813, -2.23255813, -1.48837209])
 message: 'Optimization terminated successfully'
    nfev: 37
     nit: 6
    njev: 6
  status: 0
 success: True
       x: array([-0.76744186,  0.25581395,  0.62790697, -0.11627907,  0.25581395])


Solutions

Testing at different locations yields similar values
func = 4.0930233
x = [-.7674, .2558, .6280, -.1163, .2558]